nntrainer/tensor/blas_interface.cpp

```diff
@@ -498,17 +498,30 @@ void ele_mul(const unsigned int N, const _FP16 *X, const _FP16 *Y, _FP16 *Z,
     ele_mul_fallback(N, X, Y, Z, alpha, beta, i_stride, o_stride);
 }
 
-void ele_add(const unsigned int N, const _FP16 *X, const _FP16 *Y, _FP16 *Z,
-             float alpha, float beta, unsigned int i_stride,
+void xpbypcz(const unsigned int N, const _FP16 *X, const _FP16 *Y, _FP16 *Z,
+             float beta, float gamma, unsigned int i_stride,
              unsigned int o_stride) {
   if (i_stride == 1 && o_stride == 1) {
 #if (defined USE__FP16 && USE_NEON)
-    nntrainer::neon::ele_add(N, X, Y, Z, alpha, beta);
+    nntrainer::neon::xpbypcz(N, X, Y, Z, beta, gamma);
 #else
-    ele_add_fallback(N, X, Y, Z, alpha, beta, i_stride, o_stride);
+    xpbypcz_fallback(N, X, Y, Z, beta, gamma, i_stride, o_stride);
 #endif
   } else
-    ele_add_fallback(N, X, Y, Z, alpha, beta, i_stride, o_stride);
+    xpbypcz_fallback(N, X, Y, Z, beta, gamma, i_stride, o_stride);
+}
+
+void ele_add(const unsigned int N, const _FP16 *X, const _FP16 *Y, _FP16 *Z,
+             float alpha, float beta, float gamma, unsigned int i_stride,
+             unsigned int o_stride) {
+  if (std::fpclassify(alpha) == FP_ZERO && gamma == 1.F) {
+    return saxpy(N, beta, Y, i_stride, Z, o_stride);
+  } else if (alpha == 1.F) {
```

Comment:
let's avoid using operator== for floating point numbers.

```suggestion
  if (std::fpclassify(alpha) == FP_ZERO && std::fpclassify(gamma- 1.f) == FP_ZERO) {
    return saxpy(N, beta, Y, i_stride, Z, o_stride);
  } else if (std::fpclassify(alpha - 1.f) == FP_ZERO) {
```

Reply:
Got it, but all blas-level fixes should be applied after merging #2930 unless super-urgent for now. Let me fix this afterwards.